File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public static int[] memo;
    public int solution(int n) {
        memo = new int[n + 1];

        return fibo(n) % 1234567;
    }
    private static int fibo(int n) {
        if (n == 0) {
            return memo[0];
        }
        if (n == 1) {
            memo[1] = 1;
            return memo[1];
        }
        if (memo[n] == 0) {
            memo[n] = fibo(n - 1) + fibo(n - 2);
        }
        return memo[n] % 1234567;
    }
}

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer=0
    mem=[0 for i in range(100000+1)]
    mem[1]=1
    for i in range(2, 100000+1):
        mem[i]=mem[i-2]%1234567 + mem[i-1]%1234567

    answer=mem[n] % 1234567

    return answer
